src/main/kotlin/me/spradling/gift/core/api/models/responses/RetrievedResource.kt

```diff
@@ -0,0 +1,5 @@
+package me.spradling.gift.core.api.models.responses
+
+import io.netty.handler.codec.http.HttpResponseStatus
+
+data class RetrievedResourceResponse<T> (val resource: T): ApiResponse(HttpResponseStatus.OK.code(), resource!!)
```

Comment:
The file name is `RetrievedResource.kt` but the class is `RetrievedResourceResponse`, isn't IntelliJ giving you a warning about this?

Reply:
No I wasn't getting any warnings. I got fairly confused about this too when I was trying to follow your template but this ended up working. Might be worth a refactor.